### ISSUE TITLE
Aria upgrades

### ISSUE THREAD

**Issue by humphreyja** [HarvestProfit/npk-ui#13]
`role="separator"` = `<hr />`. May want to make sure other elements have this if they are dividers like Menu.Divider

`role="tooltip"` used in conjunction with aria-describedby

`role="menu"` used for menus with a list of actions

`role="listbox"` used for a popup option that users can select one or more items from. Like the <select> tag. contains static elements with the "option" or "group" role.

`role="dialog"` used for general popups.

Search and filter sections ARIA

```html

<search> <!-- or <form role="search"> include a name="fields" to provide more details about what is being searched if there are more than one search forms -->
  <input type="search" name="search" />

 <!-- MENU -->
  <!-- Store the filter value in a hidden input type -->
  <input type="hidden" name="color" value="" />

  <!-- aria-haspopup helps point to the dialog -->
  <button type="button" aria-haspopup="dialog" aria-expanded="false" aria-controls="color-dialog" aria-label="Color Filter">
    Select Color
  </button>

   <!-- MENU END -->

</search>


<!-- Dialog to allow selecting the value -->
<div id="color-dialog" role="dialog" aria-modal="true" aria-labelledby="dialog-title" hidden>
   <!-- the aria-labelledby pointing to the header will announce to screenreaders what the dialog is. Should have at least that. -->
  <header id="dialog-title">Select a color</header>
  <button type="button" class="color-swatch" data-color="red" style="background-color: red;">Red</button>
  <button type="button" class="color-swatch" data-color="blue" style="background-color: blue;">Blue</button>
</div>

```
